### PR TITLE
Redesign Admin Dashboard

### PR DESCRIPTION
**Pull request by sampoder**
My main goal here was to clean up clutter and make it easier to read things. The main thing is the addition of a nav bar:

Screenshot 2024-05-02 at 8 34 36 PM

The other updates were made to individual pages:

- Having the "displaying X" and pagination on one row using Flexbox
- Restructuring the forms like in #4160 
- Moving the title to the nav bar

### CODE REVIEW

**Comment by maxwofford** [comment]
Ooooooh! This looks pretty! @hackclub/hcb-ops, based on the screenshot above are you cool with this change?

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -56,6 +57,43 @@ def search
     @result = users.flatten.compact
   end
 
+  def get_page_titles
+    titles = {
+      "transaction_csvs"      => "Transaction CSVs",
```
Comment (maxwofford) [inline app/controllers/admin_controller.rb]: This feels a little weird to do. Maybe something like a helper (ie. `<%= title "ACH" %>`) on each page that lets them self-identify?

**Comment by daisybanks** [comment]
Hey @sampoder! This new nav bar looks great! Super easy to read and look for things!

**Comment by sampoder** [comment]
yay!

**Comment by devramsean0** [comment]
Looks Great!

File: app/controllers/admin_controller.rb (hunk 1 above)
Comment (sampoder) [reply app/controllers/admin_controller.rb]: yeah this is jank. i think i misunderstood how the old system worked but i think i can actually get what i want using that so i'm gonna go back to that

File: app/views/admin/ach.html.erb [hunk 2]
```diff
@@ -1,24 +1,23 @@
 <% title "ACH" %>
```
Comment (sampoder) [inline app/views/admin/ach.html.erb]:
```suggestion
<% title "ACHs" %>
```

File: app/views/admin/balances.html.erb [hunk 3]
```diff
@@ -1,28 +1,27 @@
-<% title "ORGANIZATION BALANCES" %>
-
-<h1>ORGANIZATION BALANCES</h1>
-
 <%= form_with local: true, action: :balances, method: :get do |form| %>
```
Comment (sampoder) [inline app/views/admin/balances.html.erb]:
```suggestion
<% title "Organization Balances" %>

<%= form_with local: true, action: :balances, method: :get do |form| %>
```

File: app/views/admin/bank_accounts.html.erb [hunk 4]
```diff
@@ -1,8 +1,6 @@
-<% title "BANK ACCOUNTS" %>
-
-<h1>BANK ACCOUNTS</h1>
-
-<p><small><%= pluralize(@count, "result") %></small></p>
+<p class="mb2">
```
Comment (sampoder) [inline app/views/admin/bank_accounts.html.erb]:
```suggestion
<% title "Bank Accounts" %>

<p class="mb2">
```

File: app/views/admin/bank_fees.html.erb [hunk 5]
```diff
@@ -1,18 +1,15 @@
-<% title "BANK FEES" %>
-
-<h1>BANK FEES</h1>
-
-<%= paginate @bank_fees %>
-
 <%= form_with local: true, url: bank_fees_admin_index_path, method: :get do |form| %>
```
Comment (sampoder) [inline app/views/admin/bank_fees.html.erb]:
```suggestion
<% title "Bank Fees" %>

<%= form_with local: true, url: bank_fees_admin_index_path, method: :get do |form| %>
```

File: app/views/layouts/admin.html.erb [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <html>
   <head>
     <title>
-      <%= yield(:title).concat(" –") if content_for?(:title) %>
+      <%= "#{@title} –" if !@title.nil? %>
```
Comment (sampoder) [inline app/views/layouts/admin.html.erb]:
```suggestion
      <%= yield(:title).concat(" –") if content_for?(:title) %>
```

File: app/views/layouts/admin.html.erb [hunk 7]
```diff
@@ -30,56 +30,162 @@
...
-    </p>
 
-    <hr>
+      <nav>
+        <%= link_to root_path, class: "nav-item role-badge text-decoration-none badge-#{current_user.access_level}" do %>
+          <%= inline_icon "bank-account", size: 24 %>
+          <b>
+            <%= current_user.access_level %><%= ":" if !@title.nil? %>
+          </b>
+          <%= @title if !@title.nil? %>
```
Comment (sampoder) [inline app/views/layouts/admin.html.erb]:
```suggestion
          <%= yield(:title) if content_for?(:title) %>
```

File: app/views/layouts/admin.html.erb [hunk 8]
```diff
@@ -30,56 +30,162 @@
...
-      |
-      <%= link_to "stripe cards", stripe_cards_admin_index_url %>
-    </p>
 
-    <hr>
+      <nav>
+        <%= link_to root_path, class: "nav-item role-badge text-decoration-none badge-#{current_user.access_level}" do %>
+          <%= inline_icon "bank-account", size: 24 %>
+          <b>
+            <%= current_user.access_level %><%= ":" if !@title.nil? %>
```
Comment (sampoder) [inline app/views/layouts/admin.html.erb]:
```suggestion
            <%= current_user.access_level %><%= ":" if content_for?(:title) %>
```

File: app/controllers/admin_controller.rb (hunk 1 above)
Comment (sampoder) [reply app/controllers/admin_controller.rb]: i seem to have made a pig sty out of the git commit history, but jank no more

**Comment by malted** [comment]
fancy!

**Comment by sampoder** [comment]
@maxwofford @hackclub/hcb-ops would this be something worthwhile for you all? i could resolve the merge conflicts pretty easily and we could get this over the line

**Comment by hackliv** [comment]
AMAZING. Is PayPal listed on there too?

**Comment by maxwofford** [comment]
@sampoder there's your answer! yeah, please do when you have the time.

**Comment by sampoder** [comment]
@hackliv yes it is! Under `Transfers`.

**Comment by sampoder** [comment]
@maxwofford awesome! well that's been done :D

**Comment by sarthaktexas** [comment]
I don't like this

**Comment by sarthaktexas** [comment]
I will send you a figma tomorrow

**Comment by sampoder** [comment]
lol Sarthak, any specifics?

**Comment by sampoder** [comment]
@sarthaktexas Figma 👀

File: app/assets/stylesheets/admin.scss [hunk 9]
```diff
@@ -694,6 +700,199 @@ video {
...
+    display: none;
+  }
+  
+  .dropdown {
+    display: none;
+  }
+  
+  details:hover > .dropdown {
+    display: block;
+  }
```
Comment (YodaLightsabr) [inline app/assets/stylesheets/admin.scss]:
```suggestion
  summary:hover + detail, detail:hover, details:hover > .dropdown {
    display: block;
  }

  detail, .dropdown {
    display: none;
  }
```

File: app/assets/stylesheets/admin.scss [hunk 10]
```diff
@@ -694,6 +700,199 @@ video {
...
+  align-items: center;
+  gap: 4px;
+}
+
+.nav-item > .dropdown {
+  background: var(--hcb-bg-2-5);
+}
+
+.nav-item:hover {
+}
```
Comment (YodaLightsabr) [inline app/assets/stylesheets/admin.scss]:
```suggestion
```

File: app/assets/stylesheets/admin.scss [hunk 11]
```diff
@@ -694,6 +700,199 @@ video {
...
+  gap: 8px;
+}
+
+.events-filter-select > label > select {
+  flex-grow: 1
+}
+
+.events-filter-select > label > input {
+  flex-grow: 1
+}
```
Comment (YodaLightsabr) [inline app/assets/stylesheets/admin.scss]:
```suggestion
.events-filter-select > label > select, .events-filter-select > label > input {
  flex-grow: 1
}
```

**Review by YodaLightsabr — CHANGES_REQUESTED** [review]

**Review by YodaLightsabr — APPROVED** [review]

**Comment by garyhtou** [comment]
Did @hackliv or some one from operations get a chance to review this before merge?

**Comment by sampoder** [comment]
We got feedback on the early version from a bunch of people and then worked with Sarthak on later versions

**Pull request merged**